### DISCUSSION TITLE
Config file in production In Angular

### DISCUSSION THREAD

**Discussion by Dgoel515** [community/community#66870] [Programming Help]
### Body

Hello Team

can we make any config file in angular so that if we need to change the api Url we can change in Tomcat in dist folder of angular Application. 

if someone know please help. we have to come in the  main.js in dist folder or code to change the api url.if anyway to stop this please tell

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by seyLu** [comment]
#### Hi @Dgoel515,

Angular supports environment specific configurations.

```bash
# to create /src/environments/ dir which contains base config file
# environment.ts, *.developmnet.ts, *.staging.ts 
ng generate environments
```

And from there, you can modify each environment configs, for example adding apiUrl:

```diff
export const environment = {
  production: true,
+ apiUrl: 'http://my-prod-url'
};
```

and import as usual:

```ts
import { environment } from './../environments/environment';
```

or build using different config:

```bash
ng build --configuration=staging
```

You can read more on the Angular official docs: [Building and Serving Angular apps](https://angular.io/guide/build)

**Comment by vanquang202** [comment]
This doesn't quite answer your question but just want to share my view.

First, normally the API version does not change regularly. So the point of automatically changing the version here is not really reducing the effort to me.

Second, you don't want to tighten the API version that the front end uses to the back end. A new API version has a high chance of breaking the front end.

So I would say, you'll want to separate them. Separate the base URL/version with the path, do not duplicate them among APIs.

If you really want to implement the automatic versioning, just create an API to get the version on page load, then store that version in local/session storage for further usage.
